packages/material-ui/src/Menu/Menu.js

```diff
@@ -43,24 +49,56 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     classes,
     disableAutoFocusItem = false,
     MenuListProps = {},
+    // menuLevel = 1,
     onClose,
+    onEnter,
     onEntering,
+    onEntered,
     open,
     PaperProps = {},
+    setParentLastEnteredItemIndex,
     PopoverClasses,
     transitionDuration = 'auto',
+    nestedMenu = false,
     variant = 'selectedMenu',
     ...other
   } = props;
   const theme = useTheme();
 
+  const [lastEnteredItemIndex, setLastEnteredItemIndex] = useState(null);
+  const [entering, setEntering] = useState(false);
+
+  const atLeastOneNestedMenu = useMemo(() => {
+    return Array.isArray(children)
+      ? nestedMenu ||
+          children.some(
+            (child) =>
+              React.isValidElement(child) &&
+              child.props &&
+              child.props.nestedItems &&
```

Comment:
Is this initialized to null or undefined? Do we need to check its length if it exists?

Reply:
Good question. It's actually probably better that we don't since we probably want to allow either a single React component, or an array of them, to be as close as possible to the Menu children API.

I'll change this, smoke test it, and push that up.